src/RatcliffDDM.jl

```diff
@@ -1,371 +1,605 @@
 #         ν = -ν
-#         z = 1 -z
-#     end
-
-#     if st < 1.0e-3 
-#         st = 0
-#     end
-#     if sz  < 1.0e-3
-#         sz = 0
-#     end
-
-#     if sz==0
-#         if st==0 #sv=0,sz=0,st=0
-#             return pdf_sv(d, choice, rt; ϵ::Real = 1.0e-12)
-#         else #sv=0,sz=0,st=$
-#             return _simpson_1D(rt, choice, ν, η, α, z, τ, ϵ, z, z, 0, τ-st/2., τ+st/2., n_st)
-#         end
-#     else #sz=$
-#         if st==0 #sv=0,sz=$,st=0
-#             return _simpson_1D(rt, choice, ν, η, α, z, τ, ϵ, z-sz/2., z+sz/2., n_sz, τ, τ , 0)
-#         else     #sv=0,sz=$,st=$
-#             return _simpson_2D(rt, choice, ν, η, α, z, τ, ϵ, z-sz/2., z+sz/2., n_sz, τ-st/2., τ+st/2., n_st)
+#         z = 1 - z
+#         return pdf_sv(RatcliffDDM(ν, α, τ, z, η, sz, st, σ), rt; ϵ)
 #     end
+#     return pdf_sv(d, rt; ϵ)
 # end
 
-# """
-
-# _simpson_1D(x::Real, ν::Real, η::Real, α::Real, z::Real, τ::Real, ϵ::::Real, lb_z::Real, ub_z::Real, n_sz::Int, lb_t::Real, ub_t::Real, n_st::Int)
-
-#     Numerical Integration with Simpson's Method
-#     see: https://en.wikipedia.org/wiki/Simpson%27s_rule
-
-# # Arguments
-# - `x`::Real: response time
-# - `ν`::Real: response time
-# - `η`::Real: response time
-# - `α`::Real: response time
-# - `z`::Real: response time
-# - `τ`::Real: response time
-# - `ϵ`::Real: response time
-# - `lb_z`::Real: response time
-# - `ub_z`::Real: response time
-# - `n_sz`::Int: response time
-# - `lb_t`::Real: response time
-# - `ub_t`::Real: response time
-# - `n_st`::Int: response time
-
-# """
-# function _simpson_1D(x::Real, c::Real, ν::Real, η::Real, α::Real, z::Real, τ::Real, ϵ::::Real, lb_z::Real, ub_z::Real, n_sz::Int, lb_t::Real, ub_t::Real, n_st::Int)
+function _pdf_sv(d::RatcliffDDM{T}, rt::Real; ϵ::Real = 1.0e-12) where {T<:Real}
+    (ν, α, τ, z, η, sz, st, σ) = params(d)
+
+    if η == 0
+        return _pdf(SequentialSamplingModels.DDM(ν, α, τ, z), rt; ϵ)
+    end
+    # if isless(ν,0)
+    #     return pdf(SequentialSamplingModels.DDM(ν, α, τ, z), t; ϵ)  + (  ( (α*z*η)^2 - 2*ν*α*z - (ν^2)*t ) / (2*(η^2)*t+2)  ) - log(sqrt((η^2)*t+1)) + ν*α*z + (ν^2)*t*0.5
+    # end
+    # return pdf(SequentialSamplingModels.DDM(ν, α, τ, z), t; ϵ)  + (  ( (α*(1-z)*η)^2 + 2*ν*α*(1-z) - (ν^2)*t ) / (2*(η^2)*t+2)  ) - log(sqrt((η^2)*t+1)) - ν*α*(1-z) + (ν^2)*t*0.5
+    return _pdf(SequentialSamplingModels.DDM(ν, α, τ, z), rt; ϵ)  + (  ( (α*z*η)^2 - 2*ν*α*z - (ν^2)*(rt-τ) ) / (2*(η^2)*(rt-τ)+2)  ) - log(sqrt((η^2)*(rt-τ)+1)) + ν*α*z + (ν^2)*(rt-τ)*0.5
+end
+
+function pdf(d::RatcliffDDM, choice, rt; ϵ::Real = 1.0e-12)
+    if choice == 1
+        (ν, α, τ, z, η, sz, st, σ) = params(d)
+        return _pdf(RatcliffDDM(-ν, α, τ, 1-z, η, sz, st, σ), rt; ϵ)
+    end
+    return _pdf(d, rt; ϵ)
+end
+
+#use numerical integration for variability in non-decision time and bias (Ratcliff and Tuerlinckx, 2002)
+function _pdf(d::RatcliffDDM{T}, rt; ϵ::Real = 1.0e-12, n_st::Int=2, n_sz::Int=2)  where {T<:Real}
```

Comment:
For these internal functions, can you add docstrings that briefly explain their role and the meaning of the inputs?  This will help anyone who needs to debug or understand the code.